vcf.c

```diff
@@ -645,6 +650,35 @@ int bcf_hdr_parse(bcf_hdr_t *hdr, char *htxt)
         needs_sync += bcf_hdr_add_hrec(hdr, hrec);
         p += len;
     }
+
+    // Detect incorrect header lines, print a warning, and skip them.
+    // Alternatively, the program could refuse to work with such VCF. However,
+    // many VCF operations do not really care about a few malformed header
+    // lines.
+    if ( strncmp("#CHROM\tPOS",p,10) )
```

Comment:
This and the one below need to check for ```\n#CHROM\tPOS```.  By adding a line starting with ##CHROM...  I managed to core dump the parser:

```
#0  0x00007f41df84cc37 in __GI_raise (sig=sig@entry=6) at ../nptl/sysdeps/unix/sysv/linux/raise.c:56
#1  0x00007f41df850028 in __GI_abort () at abort.c:89
#2  0x000000000041dfa0 in bcf_hdr_add_sample (h=0x2441130, s=0x2441a70 "") at vcf.c:98
#3  0x000000000041e151 in bcf_hdr_parse_sample_line (h=h@entry=0x2441130, 
    str=str@entry=0x24413da "#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\tFORMAT\tX\ty\t\n##FORMAT=<ID=S,Number=1,Type=String,Description=\"Text\">\n#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\tFORMAT\tS1\tS2\tS3\n")
    at vcf.c:136
#4  0x0000000000427b38 in bcf_hdr_parse (hdr=hdr@entry=0x2441130, htxt=<optimized out>) at vcf.c:682
#5  0x000000000042820a in vcf_hdr_read (fp=0x24410a0) at vcf.c:1591
#6  0x0000000000428685 in bcf_hdr_read (hfp=<optimized out>) at vcf.c:872
#7  0x0000000000403e95 in view_vcf (in=0x24410a0, filename=0x7fff91c7c2ef "_.vcf") at htsfile.c:118
#8  0x0000000000404437 in main (argc=3, argv=0x7fff91c7a6d8) at htsfile.c:246
```